src/App.js

```diff
@@ -10,10 +11,40 @@ const tabs = [
 ];
 /* eslint-enable */
 
-const App = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedTab: tabs[0],
+  }
+
+  getTabById = id => tabs.find(tab => tab.id === id);
+
+  selectTab = (selectedTabId) => {
+    this.setState(prevState => (
+      prevState.selectedTab.id === selectedTabId
+        ? 0
+        : { selectedTab: this.getTabById(selectedTabId) }
+    ));
```

Comment:
check `selectedTab.id === selectedTabId` before calling setState, not inside if it